analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -30,6 +30,17 @@ public class ConditionEvaluatesToConstant : ConditionEvaluatesToConstantBase
 
     public override bool ShouldExecute() => true;
 
+    protected override bool IsLeftCoalesceExpression(SyntaxNode syntax) =>
+        syntax.Parent is BinaryExpressionSyntax { } binary
+        && binary.OperatorToken.IsKind(SyntaxKind.QuestionQuestionToken)
+        && binary.Left == syntax;
+
+    protected override bool IsConditionalAccessExpression(SyntaxNode syntax) =>
+        (syntax.Parent as ConditionalAccessExpressionSyntax)?.Expression == syntax;
```

Comment:
This is more fluent to read
```suggestion
        syntax.Parent is ConditionalAccessExpressionSyntax conditional && conditional.Expression == syntax;
```

Reply:
Same below, will not need the `is true`